tesseract_collision/test/collision_core_unit.cpp

```diff
@@ -27,7 +27,7 @@ TEST(TesseractCoreUnit, getCollisionObjectPairsUnit)  // NOLINT
   std::vector<tesseract_collision::ObjectPairKey> pairs =
       tesseract_collision::getCollisionObjectPairs(active_links, static_links);
 
-  EXPECT_TRUE(tesseract_common::isIdentical<tesseract_collision::ObjectPairKey>(pairs, check_pairs, false));
+  EXPECT_FALSE(tesseract_common::isIdentical<tesseract_collision::ObjectPairKey>(pairs, check_pairs, false));
```

Comment:
The isIdentical function did change, but it should be doing the same thing. Looking at the test, it seems like this should be false since acm is nullptr.

Reply:
I think this should still be returning true. It is comparing if the two vectors are equal and order does not matter which they should be equal. Am I missing something?

Reply:
The problem is that pairs is empty. getCollisionObjectPairs only pushes back pairs if acm !=nullptr. At line 28 we aren't passing it in, so it is empty. When we call it at 48, we pass in the acm, so it is the same as check_pairs.

I'm not sure why this was passing before, but the vectors are not the same. One has the pairs above. The other is empty.

Reply:
Actually, I see the reason. If the first vector is empty, nothing was being checked. I added this to the isIdentical function

Since vec1 is empty, it just returns true.
```
    return std::equal(vec1.begin(), vec1.end(), vec2.begin());
```
However, I added this up above that. So now it fails
```
  if (vec1.size() != vec2.size())
    return false;
```

Reply:
It looks like getCollisionObjectPairs is broken. If acm is nullptr then it should add the pairs. The fix is below which needs to be updated two places in the function. Can you added it to this PR?

Current:
``` c++
if (acm != nullptr && !acm(l1, l2))
        clp.push_back(tesseract_collision::getObjectPairKey(l1, l2));
```

Fixed:

``` c++
if (acm == nullptr || acm != nullptr && !acm(l1, l2))
        clp.push_back(tesseract_collision::getObjectPairKey(l1, l2));
```

Reply:
Done